src/transformState.js

```diff
@@ -5,7 +5,28 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const i of action.keysToRemove) {
+          if (state[i]) {
+            delete state[i];
+          }
+        }
+        break;
+
+      default:
```

Comment:
```suggestion
      case 'clear'
``` 
it is better to specify the сase